File: CPP/Arrays/D7_StockBuyAndSell2.cpp
#include <iostream>
#include <vector>
#include "ArrayUtils.h"
using namespace std;

int maximumProfit(const vector<int> &prices)
{
      int profit = 0;
      for (int i = 1; i < prices.size(); i++)
      {
            if (prices[i] > prices[i - 1])
            {
                  profit += prices[i] - prices[i - 1];
            }
      }
      return profit;
}

int main()
{
      vector<int> prices = {100, 180, 260, 310, 40, 535, 695};
      cout << maximumProfit(prices) << endl;
      return 0;
}

File: Javascript/Arrays/D7_StockBuyAndSell2.js
function maximumProfit(prices) {
      let profit = 0;
      for (let i = 1; i < prices.length; i++) {
            if (prices[i] > prices[i - 1]) {
                  profit += prices[i] - prices[i - 1];
            }
      }
      return profit;
}

const prices = [100, 180, 260, 310, 40, 535, 695];
console.log(maximumProfit(prices));
